frontend/src/static/js/v_authorship.js

```diff
@@ -41,23 +41,66 @@ window.vAuthorship = {
       totalBlankLineCount: '',
       filesSortType: 'lineOfCode',
       toReverseSortFiles: true,
-      filterSearch: '*',
+      searchBarValue: '',
     };
   },
 
   watch: {
-    filterType() {
-      if (this.filterType === 'checkboxes') {
-        const searchBar = document.getElementById('search');
-        searchBar.value = '';
-        this.filterSearch = '*';
-      } else {
-        this.selectedFileTypes = this.fileTypes.slice();
+    filesSortType() {
+      window.addHash('authorshipSortBy', this.filesSortType);
+      window.encodeHash();
+    },
+
+    toReverseSortFiles() {
+      window.addHash('reverseAuthorshipOrder', this.toReverseSortFiles);
+      window.encodeHash();
+    },
+
+    isLoaded() {
+      if (this.isLoaded) {
+        this.retrieveHashes();
       }
     },
   },
 
   methods: {
+    retrieveHashes() {
+      const {
+        addHash,
+        decodeHash,
+        encodeHash,
+        hashParams: hash,
+      } = window;
+
+      decodeHash();
+
+      switch (hash.authorshipSortBy) {
+      case 'path':
+      case 'fileName':
+      case 'fileType':
+        this.filesSortType = hash.authorshipSortBy;
+        break;
+      default:
+        // Invalid value, use the default value of 'lineOfCode'
+      }
+
+      this.toReverseSortFiles = hash.reverseAuthorshipOrder !== 'false';
+
+      if ('authorshipFilesGlob' in hash) {
+        this.indicateSearchBar();
+        this.searchBarValue = hash.authorshipFilesGlob;
+      } else if ('authorshipFileTypes' in hash) {
+        const parsedFileTypes = hash.authorshipFileTypes.split('~');
+        this.selectedFileTypes = parsedFileTypes.filter((type) => this.fileTypes.includes(type));
+      }
+
+      // We only add these hashes since they are propagated from summary_charts
+      addHash('tabAuthor', this.info.author);
```

Comment:
I believe the adding of hashes can be in a separate method (since it's not suited to be in a `retrieveHashes` method), and called during the `mounted` hook. I believe u added line 59-61 to avoid the adding of hashes being called before the `beforeDestroy` hook of the previous authorship component right?

Reply:
The `setInfoHash` method can be moved to its original position (below `initiate`) to only show ur addition for the method